src/libraries/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -933,5 +936,48 @@ protected virtual void GetObjectData(SerializationInfo info, StreamingContext co
         {
             throw new PlatformNotSupportedException();
         }
+
+        internal string DebuggerToString()
+        {
+            // DebuggerDisplayAttribute is inherited. Use virtual members instead of private fields to gather data.
+            int claimsCount = 0;
+            foreach (var item in Claims)
+            {
+                claimsCount++;
+            }
+
+            return $"Identity Name = {Name ?? "(null)"}, IsAuthenticated = {IsAuthenticated}, Claims Count = {claimsCount}";
+        }
+
+        private sealed class ClaimsIdentityDebugProxy
+        {
+            private readonly ClaimsIdentity _identity;
+
+            public ClaimsIdentityDebugProxy(ClaimsIdentity identity)
+            {
+                _identity = identity;
+            }
+
+            public ClaimsIdentity? Actor => _identity.Actor;
+            public string? AuthenticationType => _identity.AuthenticationType;
+            public object? BootstrapContext => _identity.BootstrapContext;
+            public IEnumerable<Claim> Claims
+            {
+                get
+                {
+                    List<Claim> claims = new List<Claim>();
+                    foreach (var c in _identity.Claims)
+                    {
+                        claims.Add(c);
+                    }
+                    return claims;
+                }
+            }
```

Comment:
```suggestion
            public IEnumerable<Claim> Claims => new List<Claim>(_identity.Claims);
```
And same question as below about typing the return of the property as `List<Claim>`.